File: lab4-records/haskell/accom.hs
module Main where

data Accommodation
    = AccHouse
      {
          acc_street :: String,
          acc_house_number :: Int,
          acc_floor_count :: Int
      }

instance (Show Accommodation) where -- how to format values of type Accommodation
    show (AccHouse street num floor_count) =
        concat
            [show floor_count,
             " storey house at ",
             show num,
             " ",
             street]

showFloorTh n
    | n == 0 = "ground"
    | lastDigit == 1  = show n ++ "st"
    | lastDigit == 2  = show n ++ "nd"
    | lastDigit == 3  = show n ++ "rd"
    | otherwise = show n ++ "th"
    where
    lastDigit = n `mod` 10

myHouse =
    AccHouse
    {
        acc_house_number = 111,
        acc_street = "Golden Avenue",
        acc_floor_count = 4
    }

myHouseB =
    AccHouse
    {
        acc_house_number = 112,
        acc_street = "Golden Avenue",
        acc_floor_count = 6
    }

{-
myFlat1 =
    AccFlat "Silver Street" 81 0

myFlat2 =
    AccFlat "Silver Street" 81 7
-}

main =
    do
--     putStr "myHouse and myHouseB are on the same street. "
--     putStrLn $ show $ sameStreet (myHouse, myHouseB)
--     putStrLn ""
    putStrLn $ show myHouse
--     putStrLn $ show myFlat1
--     putStrLn $ show myFlat2

sameStreet acc1 acc2 = -- Task 4.2.(b): fix this function
    (acc_street acc1 == acc_street acc2)

File: lab4-records/java/accom.java
public class accom
{
    
    private static class House
    {
        private String street;
        private int houseNumber;
        private int floors;

        public String getStreet() { return street; }
        public int getHouseNumber() { return houseNumber; }
        public int getFloors() { return floors; }

        public House(String street, int houseNumber, int floors)
        {
            this.street = street;
            this.houseNumber = houseNumber;
            this.floors = floors;
        }

        public String toString()
        {
            String result = "";
            result += floors;
            result += " storey house at ";
            result += getHouseNumber();
            result += " ";
            result += getStreet();

            return result;
        }
    }

    private static String floorToTh(int floor)
    {
        if (floor == 0)
        {
            return "ground";
        }
        else
        {
            switch(floor % 10)
            {
            case 1: return floor + "st";
            case 2: return floor + "nd";
            case 3: return floor + "rd";
            default: return floor + "th";
            }
        }
    }

//     private static boolean isSameStreet(Accommodation acc1, Accommodation acc2)
//     {
//         return acc1.getStreet().equals(acc2.getStreet());
//     }
    
    private static House myHouse =
        new House("Golden Avenue", 111, 4);
    
//     private static Flat myFlat1 =
//         new Flat("Silver Street", 81, 0);

//     private static Flat myFlat2 =
//         new Flat("Silver Street", 81, 7);

    public static void main(String[] args)
    {
        System.out.println(myHouse);
//         System.out.println(myFlat1);
//         System.out.println(myFlat2);

//         System.out.println("Same street? " + isSameStreet(myFlat1, myFlat2));
    }
}